engine/renderer.py

```diff
@@ -85,19 +85,24 @@ def __init__(self,
 
         ti.root.dense(ti.ij, res).place(self.color_buffer)
 
+        self.block_size = 8
+        self.block_offset = [o // self.block_size for o in self.particle_grid_offset]
         self.particle_bucket = ti.root.pointer(ti.ijk,
-                                               self.particle_grid_res // 8)
-        self.particle_bucket.dense(ti.ijk, 8).dynamic(
+                                               self.particle_grid_res // self.block_size)
+
+        self.particle_bucket.dense(ti.ijk, self.block_size).dynamic(
             ti.l, self.max_num_particles_per_cell,
             chunk_size=32).place(self.pid,
                                  offset=self.particle_grid_offset + [0])
 
-        ti.root.pointer(ti.ijk, self.particle_grid_res // 8).dense(
-            ti.ijk, 8).place(self.voxel_has_particle,
-                             offset=self.particle_grid_offset)
-        voxel_block = ti.root.pointer(ti.ijk, self.voxel_grid_res // 8)
-        voxel_block.dense(ti.ijk, 8).place(self.voxel_grid_density,
+        self.voxel_block_offset = [o // self.block_size for o in voxel_grid_offset]
+        ti.root.pointer(ti.ijk, self.particle_grid_res // self.block_size).dense(
+            ti.ijk, self.block_size).place(self.voxel_has_particle,
                                            offset=voxel_grid_offset)
+        voxel_block = ti.root.pointer(ti.ijk, self.voxel_grid_res // self.block_size)
+
+        voxel_block.dense(ti.ijk, self.block_size).place(self.voxel_grid_density,
+                                                         offset=voxel_grid_offset)
```

Comment:
Are these changes caused by the auto code formatter?

Reply:
Here I made the previously `8` to be `self.block_size` to avoid hard coding.